packages/vats/src/provisionPool.js

```diff
@@ -0,0 +1,159 @@
+
+import { fit, M, makeScalarMapStore } from '@agoric/store';
+import { E, Far } from '@endo/far';
+// TODO: move to narrower package?
+import { observeIteration, observeNotifier } from '@agoric/notifier';
+import { AmountMath, AmountShape } from '@agoric/ertp';
+import { WalletName } from '@agoric/internal';
+import { makeMyAddressNameAdmin, PowerFlags } from './core/basic-behaviors.js';
+
+const { details: X, quote: q } = assert;
+
+const privateShape = harden({
+  poolBank: M.eref(M.remotable('bank')),
+});
+
+/**
+ * Given attenuated access to the funding purse,
+ * handle requests to provision smart wallets.
+ *
+ * @param {(address: string, depositFacet: ERef<DepositFacet>) => Promise<void>} sendInitialPayment
+ */
+const makeBridgeProvisionTool = sendInitialPayment => {
+  /**
+   * @param {{
+   *   bankManager: ERef<BankManager>,
+   *   namesByAddressAdmin: ERef<NameAdmin>,
+   *   walletFactory: ERef<import('@agoric/vats/src/core/startWalletFactory').WalletFactoryStartResult['creatorFacet']>
+   * }} io
+   */
+  const makeHandler = ({ bankManager, namesByAddressAdmin, walletFactory }) =>
+    Far('provisioningHandler', {
+      fromBridge: async (_srcID, obj) => {
+        assert(namesByAddressAdmin, 'no namesByAddressAdmin');
+        assert.equal(
+          obj.type,
+          'PLEASE_PROVISION',
+          X`Unrecognized request ${obj.type}`,
+        );
+        const { address, powerFlags } = obj;
+        console.info('PLEASE_PROVISION', address, powerFlags);
+
+        if (!powerFlags.includes(PowerFlags.SMART_WALLET)) {
+          return;
+        }
+
+        const { nameHub, myAddressNameAdmin } = makeMyAddressNameAdmin(address);
+
+        const bank = E(bankManager).getBankForAddress(address);
+        await Promise.all([
+          E(namesByAddressAdmin).update(address, nameHub, myAddressNameAdmin),
+          E(walletFactory).provideSmartWallet(
+            address,
+            bank,
+            myAddressNameAdmin,
+          ),
+          sendInitialPayment(address, nameHub.lookup(WalletName.depositFacet)),
+        ]);
+
+        console.info('provisioned', address, powerFlags);
+      },
+    });
+  return makeHandler;
+};
+
+/**
+ * @typedef {StandardTerms & {
+ *   perAccountInitialAmount: Amount<'nat'>,
+ * }} ProvisionTerms
+ *
+ * @param {ZCF<ProvisionTerms>} zcf
+ * @param {{
+ *   poolBank: import('@endo/far').FarRef<import('./vat-bank.js').Bank>,
+ * }} privateArgs
+ * @returns
+ */
+export const start = (zcf, privateArgs) => {
+  // TODO: make initial amount governed
+  const { perAccountInitialAmount } = zcf.getTerms();
+  fit(privateArgs, privateShape, 'provisionPool privateArgs');
+  const { poolBank } = privateArgs;
+  fit(perAccountInitialAmount, AmountShape, 'perAccountInitialAmount');
+  const { brand: poolBrand } = perAccountInitialAmount;
+  /** @type {ERef<Purse>} */
+  // @ts-expect-error vbank purse is close enough for our use.
+  const fundPurse = E(poolBank).getPurse(poolBrand);
```

Comment:
did you run accross this, @turadg ? any clues?

Reply:
yes. VirtualPurse is what the bank returns and it's slightly different. For example, `withdraw` is sync on Purse and async on VirtualPurse. @michaelfig documented this, https://github.com/Agoric/agoric-sdk/blob/28b104b76bef2ef1683cdefee3e89183ecff39b0/packages/vats/test/test-vat-bank.js#L135-L142

Making ERTP methods accept remote payments seems worth doing soon. WDYT @michaelfig ?